### PR TITLE
Investigate opta ingester

### PR DESCRIPTION
**Pull request by dragospitica**

### CODE REVIEW

**Comment by AdamDM** [comment]
lgtm

File: src/main/java/org/atlasapi/remotesite/events/EventsIngestTask.java [hunk 1]
```diff
@@ -16,7 +16,9 @@
     private final EventsFetcher<S, T, M> fetcher;
     private final DataHandler<S, T, M> dataHandler;
     
-    public EventsIngestTask(Logger log, EventsFetcher<S, T, M> fetcher, 
+    public EventsIngestTask(
+            Logger log,
+            EventsFetcher<S, T, M> fetcher,
             DataHandler<S, T, M> dataHandler) {
```
Comment (jamiesprax) [inline src/main/java/org/atlasapi/remotesite/events/EventsIngestTask.java]: ) { on new line seeing as you're reformatting anyway

File: src/main/java/org/atlasapi/remotesite/opta/events/sports/OptaSportsDataHandler.java [hunk 2]
```diff
@@ -140,12 +141,9 @@ public OptaSportsDataHandler(OrganisationStore organisationStore, EventStore eve
...
     private String getVenueData(List<SportsStats> stats) {
-        return Iterables.getOnlyElement(Iterables.filter(stats, new Predicate<SportsStats>() {
-            @Override
-            public boolean apply(SportsStats input) {
-                return VENUE_TYPE.equals(input.attributes().type());
-            }
-        })).value();
+        return Iterables.getOnlyElement(stats.stream()
+                .filter(input -> VENUE_TYPE.equals(input.attributes().type()))
+                .collect(Collectors.toList())).value();
```
Comment (jamiesprax) [inline src/main/java/org/atlasapi/remotesite/opta/events/sports/OptaSportsDataHandler.java]: You can just use a findFirst() and then an orElse throw missing resource exception to save the Iterables and collecting to a list.

File: src/main/java/org/atlasapi/remotesite/opta/events/sports/OptaSportsDataHandler.java [hunk 3]
```diff
@@ -155,7 +153,9 @@ public Organisation apply(SportsTeamData input) {
                 return getTeamByUri(utility.createTeamUri(sportType, input.attributes().teamRef())).orNull();
             }
         });
-        return Iterables.filter(organisations, Predicates.notNull());
+        return StreamSupport.stream(organisations.spliterator(), false)
+                .filter(Predicates.notNull()::apply)
```
Comment (jamiesprax) [inline src/main/java/org/atlasapi/remotesite/opta/events/sports/OptaSportsDataHandler.java]: You should use the java 8 `.filter(Objects::notNull)`

**Pull request merged**